### PR TITLE
Hopefully fix vmmap recursion issues

### PR DESCRIPTION
**Pull request by disconnect3d**
This fixes the issue from #1565 but does not fix all vmmap/ehdr recursion errors.

### CODE REVIEW

File: pwndbg/gdblib/vmmap.py [hunk 1]
```diff
@@ -88,13 +87,25 @@ def get() -> Tuple[pwndbg.lib.memory.Page, ...]:
...
-        and pwndbg.gdblib.arch.current in ("i386", "x86-64", "aarch64", "riscv:rv64")
+    if is_corefile():
+        return tuple(coredump_maps())
+
+    proc_maps = proc_pid_maps()
+
+    # The `proc_maps` is usually a tuple of Page objects but it can also be:
+    #   None    - when /proc/$pid/maps does not exist/is not available
+    #   tuple() - when the process has no maps yet which happens only during its very early init
+    #             (usually when we attach to a process)
```
Comment (gsingh93) [inline pwndbg/gdblib/vmmap.py:99]: `proc_pid_maps` currently describes the return value as "A list of pwndbg.lib.memory.Page objects". We should update it with this information.

File: pwndbg/gdblib/vmmap.py [hunk 2]
```diff
@@ -88,13 +87,25 @@ def get() -> Tuple[pwndbg.lib.memory.Page, ...]:
...
+    if proc_maps is not None:
+        return proc_maps
+
+    pages = []
+    if pwndbg.gdblib.qemu.is_qemu_kernel() and pwndbg.gdblib.arch.current in (
```
Comment (gsingh93) [inline pwndbg/gdblib/vmmap.py:104]: Doesn't need to be done in this PR, but was just thinking this function would be more readable if we move the body of this if statement to some function like `qemu_kernel_maps`. Then it's more clear what's going on in this function, i.e. first coredump maps, then process maps, then qemu kernel maps, etc.

File: pwndbg/gdblib/vmmap.py [hunk 3]
```diff
@@ -112,9 +123,6 @@ def get() -> Tuple[pwndbg.lib.memory.Page, ...]:
         elif kernel_vmmap == "monitor":
             pages.extend(kernel_vmmap_via_monitor_info_mem())
 
-    if not pages and is_corefile():
-        pages.extend(coredump_maps())
-
     # TODO/FIXME: Do we still need it after coredump_maps()?
     # Add tests for other cases and see if this is needed e.g. for QEMU user
     # if not, remove the code below & cleanup other parts of Pwndbg codebase
```
Comment (gsingh93) [inline pwndbg/gdblib/vmmap.py]: I guess if this really isn't needed, just removing it would fix the recursion issues as well?

**Review by gsingh93 — APPROVED** [review]
lgtm

File: pwndbg/gdblib/vmmap.py (hunk 1 above)
Comment (disconnect3d) [reply pwndbg/gdblib/vmmap.py:99]: Fixed

File: pwndbg/gdblib/vmmap.py (hunk 3 above)
Comment (disconnect3d) [reply pwndbg/gdblib/vmmap.py]: I rephrased this. I probably need to test this edge case a bit more.
